### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by najibullahjafari**
Here:

- We created a class called Solver.
- We created a method called factorial that takes one argument, an integer N, and returns the [factorial]
- We created a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
- We created a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
- [ ] When N is divisible by 3, return "fizz".
- [ ] When N is divisible by 5, return "buzz".
- [ ] When N is divisible by 3 and 5, return "fizzbuzz".
- [ ] Any other case, return N as a string (e.g. say N is 7 then return "7").

### CODE REVIEW

File: solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      f = @solver.fizzbuzz(15)
+      expect(f).to eq('fizzbuzz')
+    end
+  end
+
+  describe 'fizzbuzz' do
+    it 'returns a string for a number that is not divisible by 3 and 5' do
+      f = @solver.fizzbuzz(13)
+      expect(f).to be_a(String)
+    end
```
Comment (Efiamotu-1) [inline solver_spec.rb]: - [ ] This method should rather return whatever number was inputed. see the example code snippet below :+1: 
```ruby
expect(resolver.fizzbuzz(7)).to eql('7')  # N is not divisible by 3 or 5
```

File: solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  context 'Factorial Method' do
+    it 'Factorial Number Positive' do
+      result = @solver.factorial(2)
+      expect(result).to eq(2)
+    end
+
+    it 'Factorial Number Negative' do
+      expect { @solver.factorial(-2) }.to raise_error(ArgumentError, 'Factorial is not defined for negative integers.')
+    end
+  end
```
Comment (Efiamotu-1) [inline solver_spec.rb]: - [ ] You have not added the method to test for the Special case - when input is 0, the result should equal 1. Please kindly add this test case and resubmit for approval :+1:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @najibullahjafari & @MasumaJaffery,  

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Efiamotu-1** [comment]
- [ ] Please kindly edit the title of your project in the readme file :+1: 
pr

**Review by BregornOriginal — APPROVED** [review]
Hi @Team

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 

giphy

## Highlights

✅ Linters are passing
✅ Excellent commit history
✅ Good work implementing TDD technique! 🥇 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**